src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -175,10 +191,36 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         logger.info(`Leaving call session in room ${this.room.roomId}`);
         this.relativeExpiry = undefined;
         this.activeFoci = undefined;
+        this.encryptMedia = false;
         this.emit(MatrixRTCSessionEvent.JoinStateChanged, false);
         this.updateCallMembershipEvent();
     }
 
+    /**
+     * Re-sends the encryption key room event with a new key
+     */
+    private async updateEncryptionKeyEvent(): Promise<void> {
+        if (!this.isJoined()) return;
+        if (!this.encryptMedia) return;
+
+        logger.info(`MatrixRTCSession updating encryption key for room ${this.room.roomId}`);
+
+        const encryptionKey = getNewEncryptionKey();
+        await this.client.sendEvent(this.room.roomId, EventType.CallEncryptionPrefix, {
+            "m.encryption_key": encryptionKey,
+            "m.device_id": this.client.getDeviceId(),
+        } as EncryptionKeyEventContent);
```

Comment:
Feels like this probably ought to include the call ID (ie. empty string) & perhaps scope too? Enough to identify what call it's for, anyway.

Reply:
Good point!